### ISSUE TITLE
Review legacy Security COMPONENTS for migration

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4003]
Related issues:
- **[Review legacy Security OVERVIEW (Read me first)](https://github.com/carbon-design-system/ibm-products/issues/4052)**
- Review legacy Security COMPONENTS for migration `You are here`
- [Review legacy Security PATTERNS for migration](https://github.com/carbon-design-system/ibm-products/issues/4036)
- [Review legacy Security LAYOUTS for migration](https://github.com/carbon-design-system/ibm-products/issues/4037)

_This is a summary report. The details listed for each component is not comprehensive._

More details provided with each component report below.

**"Easy" to migrate.** Has an up-to-date code spec similar to **Carbon for IBM Products**.

- DelimitedList
- ICA

**"Difficult" to migrate.** Must be converted from a `class` to a `function`.

- Nav
- TruncatedList

**"Difficult" to migrate.** There are three versions to be considered before we even start.

- (Data)Decorator

**Not recommended to be migrated.** See each item for details.

- StackedNotification
- StatusIcon
- StringFormatter
- TimeIndicator
- TypeLayout

---

### Decorator and DataDecorator

This needs to be sorted out before we start: is it possible to use any _one_ as a single source of truth?

**Security** Decorator (component): [GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/DataDecorator/Decorator/Decorator.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-decorator--default)
- **✓** Follows **Carbon for IBM Products** component architecture.

**Security** DataDecorator (pattern): [GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/DataDecorator/DataDecorator.js), [Storybook](https://ibm-security.carbondesignsystem.com/?path=/story/components-decorator--default)
- **✗** See `This component must be refactored from a Class to a Function`. ([Link](https://github.com/carbon-design-system/ibm-products/issues/4052))
- **✗** Is importing a `Panel` and other components.

**Data Explorer** Decorator (component): [GitHub](https://github.ibm.com/investigate/investigate/blob/develop/src/components/Decorator/Decorator.js), [Storybook](https://pages.github.ibm.com/investigate/investigate/?path=/story/decorator--default)
- **✗** Is tightly-coupled to the CP4S Shell.

---

### DelimitedList

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/DelimitedList/DelimitedList.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-delimitedlist--default)

#### TLDR

- **✓** Follows **Carbon for IBM Products** component architecture.

---

### ICA

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/ICA/ICA.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-ica--default)

#### TLDR

- **✓** Follows **Carbon for IBM Products** component architecture.
- **✗** `isDevelopment()` ?

#### PROBLEMATIC IMPORTS

```
import 'numeral/locales';
import Icon from '../Icon/Icon';
import Tooltip from '../Tooltip';
import isDevelopment from '../../globals/env';
```

---

### Nav

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/Nav/Nav.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-nav--default)

#### TLDR

- **✗** See `This component must be refactored from a Class to a Function`. ([Link](https://github.com/carbon-design-system/ibm-products/issues/4052))

#### PROBLEMATIC IMPORTS

Importing global variables, functions, methods, utilities, etc.

```
import window from '../../globals/utils/globalRoot';
```

---

### StackedNotification

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/StackedNotification/StackedNotification.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-stackednotification--default)

Skipping code review; as per Box note, this may be integrated as a variant of [Carbon Notifications](https://react.carbondesignsystem.com/?path=/story/components-notifications-actionable--default).

---

### StatusIcon

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/StatusIcon/StatusIcon.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-statusicon--default)

Skipping code review; as per Box note, this may be rolled up into [Carbon Status Indicator Pattern](https://carbondesignsystem.com/patterns/status-indicator-pattern/).

---

### StringFormatter

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/StringFormatter/StringFormatter.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-stringformatter--default)

#### TLDR

- **✓** Follows **Carbon for IBM Products** component architecture.
- **✗** _Not recommended as a component; prefer a pattern if necessary._

#### Notes

- There are zero IBM Design considerations for this component; it's just truncating text.
- This seems like an awful lot of work for a component that essentially applies two CSS _properties_ (not classes) to a span, with a `<tooltip>`.

---

### TimeIndicator

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/TimeIndicator/index.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-timeindicator--default)

#### TLDR

- **✓** Follows **Carbon for IBM Products** component architecture.
- **✗** _Not recommended as a component; prefer a pattern if necessary._

#### Notes

- This seems like an awful lot of work for a component that essentially applies a Carbon font-family class to a span, with an `<icon>`.

---

### TruncatedList

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/TruncatedList/TruncatedList.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-truncatedlist--default)

#### TLDR

- **✗** See `This component must be refactored from a Class to a Function`. ([Link](https://github.com/carbon-design-system/ibm-products/issues/4052))

---

### TypeLayout

[GitHub](https://github.com/carbon-design-system/ibm-products/blob/main_v1/packages/security/src/components/TypeLayout/TypeLayout.js), [Storybook](https://v1-ibm-products.carbondesignsystem.com/?path=/story/security-components-typelayout--default)

#### TLDR

- **✗** _Not recommended, it's simply importing components as-is from `Carbon`._ Prefer importing `StructuredList` directly from `Carbon` instead. ([GitHub](https://github.com/carbon-design-system/carbon/blob/main/packages/react/src/components/StructuredList/StructuredList.tsx), [Storybook](https://react.carbondesignsystem.com/?path=/story/components-structuredlist--default))

#### PROBLEMATIC IMPORTS

Importing other components.

```
import {
  StructuredListBody,
  StructuredListCell,
  StructuredListRow,
  StructuredListWrapper,
} from '../StructuredList';
```

_Security's_ `StructuredList` is simply re-exporting components from `Carbon`. 

```
export {
  StructuredListBody,
  StructuredListCell,
  StructuredListHead,
  StructuredListInput,
  StructuredListRow,
  StructuredListSkeleton,
  StructuredListWrapper,
} from 'carbon-components-react';
```

**Comment by elycheea** [comment]
Closing for now since it’s documented as part of the Security migration epic.
